### ISSUE TITLE
Ability to create tasks in PullRequest

### ISSUE THREAD

**Issue by wiibaa** [cdancy/bitbucket-rest#101]
hello,
I am searching for a lib to use as a bot for our bitbucket instance as one could use octokit  with github.
My use case is to scan for newly created pull request and create a comment that hold a list of tasks to be performed by the reviewer(s).
From what I could see, you already have the PR and Commetns APIs but nothing about tasks, am I right ?
Is it already on your roadmap or shall it wait for external contribution ?

Big :+1:  for the clean organisation of the repo and clear release note that helped me to look around quiet efficiently 

REST method I'm looking to use: 
POST /rest/api/1.0/tasks
from https://developer.atlassian.com/static/rest/bitbucket-server/latest/bitbucket-rest.html

**Comment by cdancy** [comment]
@wiibaa glad to hear you like the project! We do indeed have a PullRequest and Comments API. I've not coded anything up for the `tasks` endpoint but if we did, that would fall under the PullRequest api. I can surely whip something together for you if you just need that one endpoint. I am curious though looking over the documentation as I see the below in the response from bitbucket. Do we know what is supposed to be in either of these lists? I could venture a guess for the comments but not for the tasks. I assume the `tasks` are what you'd be looking for? Do have an example of what this is supposed to look like? Sometimes their documentation can leave a bit to be desired.

```
                "comments": [],
                "tasks": [],
```

**Comment by wiibaa** [comment]
@cdancy  thanks for the positive feedback; to provide more accurate inputs I made a manual quick test

here is what a call to `http://example.com/rest/api/1.0/projects/{projectKey}/repos/{repositorySlug}/pull-requests/{pullRequestId}/tasks` would look like for a PR that contains 2 tasks in a single comment.

https://gist.github.com/wiibaa/57907ab3ee9fe6161e5037c2ff2ac95d

I also quickly tried posting a new task into a comment through the API and collected the request/response also in the gist.

HTH

**Comment by cdancy** [comment]
Ok. Never used this feature actually. Got a good understanding of how it works now. I can probably knock this out for you tomorrow and do a new release within the next few days.

**Comment by wiibaa** [comment]
@cdancy I suppose your [last comment](https://github.com/cdancy/bitbucket-rest/issues/102#issuecomment-312934738) belongs here, and yes my idea was to auto-create a set of tasks that are required for all PR, so the POST endpoint is also needed.

**Comment by wiibaa** [comment]
@cdancy thanks for the fast work on this topic. I'm testing with looping on ALL existing PR in our repository and I stumbled upon a missed case 

`SEVERE: Error parsing input: No enum constant com.cdancy.bitbucket.rest.domain.activities.Activities.ActivitiesType.MERGED`

**Comment by cdancy** [comment]
@wiibaa I can slip that in as well.

**Comment by cdancy** [comment]
@wiibaa changes have been committed to master. Would you care to build things locally and see if all cases are covered for you? Also ... added a new api `TasksApi` which will cover all task related operations. Currently there is a single endpoint `create`. Let me know how things work out for you.

https://github.com/cdancy/bitbucket-rest/blob/master/src/main/java/com/cdancy/bitbucket/rest/features/TasksApi.java#L48

**Comment by wiibaa** [comment]
@cdancy AWESOME. Creating a task work like a charm. Feels dizzy to only require the commentId, but is working fine. Thanks you

**Comment by cdancy** [comment]
Closing as this functionality was merged in #104
